Unlink all accounts on keystore  invalidation

Otherwise, it raises an exception:

```
Caused by: javax.crypto.BadPaddingException
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:515)
        at javax.crypto.Cipher.doFinal(Cipher.java:2055)
        at io.horizontalsystems.bankwallet.core.security.CipherWrapper.decrypt(CipherWrapper.kt:40)
        at io.horizontalsystems.bankwallet.core.security.EncryptionManager.decrypt(EncryptionManager.kt:17)
        at io.horizontalsystems.bankwallet.core.storage.DatabaseConverters.decryptSecretList(DatabaseConverters.kt:78)
        at io.horizontalsystems.bankwallet.core.storage.AccountsDao_Impl.getDeleted(AccountsDao_Impl.java:391)
        at io.horizontalsystems.bankwallet.core.storage.AccountsStorage.getDeleted(AccountsStorage.kt:44)
        at io.horizontalsystems.bankwallet.core.managers.AccountManager$clearAccounts$clearAsync$1.call(AccountManager.kt:70)
        at io.horizontalsystems.bankwallet.core.managers.AccountManager$clearAccounts$clearAsync$1.call(AccountManager.kt:14)
        at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
        at io.reactivex.Single.subscribe(Single.java:3603) 
        at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onSuccess(SingleFlatMap.java:84) 
        at io.reactivex.internal.operators.single.SingleTimer$TimerDisposable.run(SingleTimer.java:56) 
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) 
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
     Caused by: android.security.KeyStoreException: Invalid argument
        at android.security.KeyStore.getKeyStoreException(KeyStore.java:839)
        at android.security.keystore.KeyStoreCryptoOperationChunkedStreamer.doFinal(KeyStoreCryptoOperationChunkedStreamer.java:224)
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineDoFinal(AndroidKeyStoreCipherSpiBase.java:506)
        at javax.crypto.Cipher.doFinal(Cipher.java:2055) 
        at io.horizontalsystems.bankwallet.core.security.CipherWrapper.decrypt(CipherWrapper.kt:40) 
        at io.horizontalsystems.bankwallet.core.security.EncryptionManager.decrypt(EncryptionManager.kt:17) 
        at io.horizontalsystems.bankwallet.core.storage.DatabaseConverters.decryptSecretList(DatabaseConverters.kt:78) 
        at io.horizontalsystems.bankwallet.core.storage.AccountsDao_Impl.getDeleted(AccountsDao_Impl.java:391) 
        at io.horizontalsystems.bankwallet.core.storage.AccountsStorage.getDeleted(AccountsStorage.kt:44) 
        at io.horizontalsystems.bankwallet.core.managers.AccountManager$clearAccounts$clearAsync$1.call(AccountManager.kt:70) 
        at io.horizontalsystems.bankwallet.core.managers.AccountManager$clearAccounts$clearAsync$1.call(AccountManager.kt:14) 
        at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44) 
        at io.reactivex.Single.subscribe(Single.java:3603) 
        at io.reactivex.internal.operators.single.SingleFlatMap$SingleFlatMapCallback.onSuccess(SingleFlatMap.java:84) 
        at io.reactivex.internal.operators.single.SingleTimer$TimerDisposable.run(SingleTimer.java:56) 
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38) 
        at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:764) 
```

--- a/app/src/main/java/io/horizontalsystems/bankwallet/core/Interfaces.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/core/Interfaces.kt
@@ -4,7 +4,6 @@ import android.text.SpannableString
 import androidx.core.hardware.fingerprint.FingerprintManagerCompat
 import com.google.gson.JsonObject
 import io.horizontalsystems.bankwallet.core.managers.ServiceExchangeApi.HostType
-import io.horizontalsystems.bankwallet.core.storage.AccountRecord
 import io.horizontalsystems.bankwallet.entities.*
 import io.horizontalsystems.bankwallet.entities.Currency
 import io.horizontalsystems.bankwallet.lib.chartview.ChartView
@@ -347,7 +346,7 @@ interface IAccountsStorage {
     fun delete(id: String)
     fun getNonBackedUpCount(): Flowable<Int>
     fun clear()
-    fun getDeleted(): List<AccountRecord>
+    fun getDeletedAccountIds(): List<String>
     fun clearDeleted()
 }
 

--- a/app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AccountCleaner.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AccountCleaner.kt
@@ -1,22 +1,21 @@
 package io.horizontalsystems.bankwallet.core.managers
 
 import io.horizontalsystems.bankwallet.core.adapters.*
-import io.horizontalsystems.bankwallet.core.storage.AccountRecord
 
 class AccountCleaner(private val testMode: Boolean) {
 
-    fun clearAccounts(accountRecords: List<AccountRecord>) {
+    fun clearAccounts(accountRecords: List<String>) {
         accountRecords.forEach { clearAccount(it) }
     }
 
-    private fun clearAccount(accountRecord: AccountRecord) {
-        BinanceAdapter.clear(accountRecord.id, testMode)
-        BitcoinAdapter.clear(accountRecord.id, testMode)
-        BitcoinCashAdapter.clear(accountRecord.id, testMode)
-        DashAdapter.clear(accountRecord.id, testMode)
-        EosAdapter.clear(accountRecord.id, testMode)
-        EthereumAdapter.clear(accountRecord.id, testMode)
-        Erc20Adapter.clear(accountRecord.id, testMode)
+    private fun clearAccount(accountId: String) {
+        BinanceAdapter.clear(accountId, testMode)
+        BitcoinAdapter.clear(accountId, testMode)
+        BitcoinCashAdapter.clear(accountId, testMode)
+        DashAdapter.clear(accountId, testMode)
+        EosAdapter.clear(accountId, testMode)
+        EthereumAdapter.clear(accountId, testMode)
+        Erc20Adapter.clear(accountId, testMode)
     }
 
 }

--- a/app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AccountManager.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/core/managers/AccountManager.kt
@@ -67,7 +67,7 @@ class AccountManager(private val storage: IAccountsStorage, private val accountC
 
     override fun clearAccounts() {
         val clearAsync = Single.fromCallable {
-            accountCleaner.clearAccounts(storage.getDeleted())
+            accountCleaner.clearAccounts(storage.getDeletedAccountIds())
             storage.clearDeleted()
         }
 

--- a/app/src/main/java/io/horizontalsystems/bankwallet/core/storage/AccountsDao.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/core/storage/AccountsDao.kt
@@ -18,8 +18,8 @@ interface AccountsDao {
     @Query("SELECT * FROM AccountRecord WHERE deleted = 0")
     fun getAll(): List<AccountRecord>
 
-    @Query("SELECT * FROM AccountRecord WHERE deleted = 1")
-    fun getDeleted(): List<AccountRecord>
+    @Query("SELECT id FROM AccountRecord WHERE deleted = 1")
+    fun getDeletedIds(): List<String>
 
     @Query("SELECT COUNT(*) FROM AccountRecord WHERE isBackedUp = 0 AND deleted = 0")
     fun getNonBackedUpCount(): Flowable<Int>

--- a/app/src/main/java/io/horizontalsystems/bankwallet/core/storage/AccountsStorage.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/core/storage/AccountsStorage.kt
@@ -40,8 +40,8 @@ class AccountsStorage(appDatabase: AppDatabase) : IAccountsStorage {
                 }
     }
 
-    override fun getDeleted(): List<AccountRecord> {
-        return dao.getDeleted()
+    override fun getDeletedAccountIds(): List<String> {
+        return dao.getDeletedIds()
     }
 
     override fun clearDeleted() {

--- a/app/src/main/java/io/horizontalsystems/bankwallet/modules/keystore/KeyStorePresenter.kt
+++ b/app/src/main/java/io/horizontalsystems/bankwallet/modules/keystore/KeyStorePresenter.kt
@@ -7,11 +7,15 @@ class KeyStorePresenter(private val interactor: KeyStoreModule.IInteractor,
     var view: KeyStoreModule.IView? = null
 
     override fun viewDidLoad() {
-        interactor.resetApp()
-
         when (mode) {
-            KeyStoreModule.ModeType.NoSystemLock -> view?.showNoSystemLockWarning()
-            KeyStoreModule.ModeType.InvalidKey -> view?.showInvalidKeyWarning()
+            KeyStoreModule.ModeType.NoSystemLock -> {
+                interactor.resetApp()
+                view?.showNoSystemLockWarning()
+            }
+            KeyStoreModule.ModeType.InvalidKey -> {
+                interactor.resetApp()
+                view?.showInvalidKeyWarning()
+            }
             KeyStoreModule.ModeType.UserAuthentication -> view?.promptUserAuthentication()
         }
     }

